### PR TITLE
Add security section in settings screen

### LINKED ISSUE #310: [UI] Settings Screen - Security section
- [x] Show Primary Password
- [x] Show Primary OOB OTP
- [x] Show Secondary TOTP
- [x] Show Secondary OOB OTP
- [x] Show Secondary Password
- [x] Show Recovery code
- [x] Show revoke all device tokens button

### PR DESCRIPTION
**Pull request by louischan-oursky**
fix #310 

Most buttons are not functional. They will be implemented by later PRs.

We also discovered a problem with our localization system. We cannot localize date time (yet). ICU MessageFormat has support for datetime formatting. But our MessageFormat does not support it (yet).

### CODE REVIEW

File: pkg/auth/handler/webapp/settings.go [hunk 1]
```diff
@@ -26,19 +31,72 @@ func ConfigureSettingsRoute(route httproute.Route) httproute.Route {
 		WithPathPattern("/settings")
 }
 
+type SettingsViewModel struct {
+	Authenticators           []*authenticator.Info
+	RecoveryCodes            []*mfa.RecoveryCode
```
Comment (kiootic) [inline pkg/auth/handler/webapp/settings.go:36]: I don't think we should load all recovery codes upon entering settings page: viewing recovery code logs a security event (which GitHub does), so it should be loaded only on the page specific to viewing recovery code.

File: pkg/auth/handler/webapp/settings.go [hunk 2]
```diff
@@ -26,19 +31,72 @@ func ConfigureSettingsRoute(route httproute.Route) httproute.Route {
...
+	SecondaryTOTPEnabled     bool
```
Comment (kiootic) [inline pkg/auth/handler/webapp/settings.go:37]: It seems there are used to determine whether the relevant sections should be shown. However, if user has a secondary authenticator not enabled in config (i.e. config changed), should we still show that section?

File: templates/auth_ui_settings.html [hunk 3]
```diff
@@ -12,13 +12,201 @@
...
+    </p>
+    <a class="action" href="/settings/identity">
+      {{ template "details-button-label" }}
+    </a>
+  </section>
+  {{ end}}{{ end }}
+  {{ end }}
+
+  <!-- MFA title -->
+  <section class="settings-row settings-page-section-with-title">
```
Comment (kiootic) [inline templates/auth_ui_settings.html:77]: It seems it still show when no secondary authenticator is configured. We may want to hide it if the section is empty?

File: templates/auth_ui_settings.html [hunk 4]
```diff
@@ -12,13 +12,201 @@
...
+      {{ template "settings-page-recovery-code-description" }}
+    </p>
+    <a class="action" href="#">
+      {{ template "show-recovery-code-button-label" }}
+    </a>
+  </section>
+  {{ end }}
+
+  <!-- Revoke all device tokens -->
+  {{ if $.RecoveryCodes }}
```
Comment (kiootic) [inline templates/auth_ui_settings.html:192]: It seems we're assuming 'have recovery codes' = 'MFA enabled'. I suppose we should lift the 'MFA enabled' flag to a field in view model to be more explicit.

**Comment by louischan-oursky** [comment]
@kiootic updated.

**Review by kiootic — APPROVED** [review]

**Pull request merged**